scripts/e2e/compare_metrics.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2024 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import re
+import sys
+from pathlib import Path
+
+def read_metric_file(file_path):
+    with open(file_path, 'r') as f:
+        return f.readlines()
+
+def parse_metric_line(line):
+
+    # Skip empty lines or comments
+    line = line.strip()
+    if not line or line.startswith('#'):
+        return None
+    
+    # Extract metric name and labels section
+    match = re.match(r'([^{]+)(?:{(.+)})?(?:\s+[-+]?[0-9.eE+-]+)?$', line)
+    if not match:
+        return None
+        
+    name = match.group(1)
+    labels_str = match.group(2) or ''
+    
+    # Parse labels into a frozenset of (key, value) tuples
+    labels = []
+    if labels_str:
+        for label in labels_str.split(','):
+            label = label.strip()
+            if '=' in label:
+                key, value = label.split('=', 1)
+                labels.append((key.strip(), value.strip()))
+    
+    # Sort labels and convert to frozenset for comparison
+    return (name, frozenset(sorted(labels)))
+
+def generate_diff(file1_lines, file2_lines, file1_name, file2_name):
+
+    # Parse metrics from both files
+    metrics1 = {}  # {(name, labels_frozenset): original_line}
+    metrics2 = {}
+    
+    for line in file1_lines:
+        parsed = parse_metric_line(line)
+        if parsed:
+            metrics1[parsed] = line.strip()
+            
+    for line in file2_lines:
+        parsed = parse_metric_line(line)
+        if parsed:
+            metrics2[parsed] = line.strip()
+    
+    # Generate diff
+    diff_lines = []
```

Comment:
aren't there some existing & well maintained libraries, like deepdiff, that can do this automatically? I don't want us to maintain unnecessary code.